nilearn/region_extraction/regionextractor.py

```diff
@@ -0,0 +1,341 @@
+        switches between which technique to be used. If
+        'voxel_regions', regions are first thresholded to
+        foreground objects and then labels are assigned to
+        each objects to get the end result. If 'local_regions',
+        same strategy follows as in 'voxel_regions' but the
+        labels are assigned after each region is detected by
+        its own peak max value technically called as seed points.
+
+    smooth_fwhm: scalar, default smooth_fwhm=6. optional
+        smoothing parameter as a full width half maximum,
+        in millimetres. This scalar value is applied on
+        all three directions.
+
+    Attributes
+    ----------
+    `regions_`: Niimg-like image/object
+        a list of seperate regions with each region lying on
+        a 3D volume concatenated into a 4D Nifti object.
+
+    `index_`: a numpy array
+        an array of list of indices where each index value is
+        assigned to each seperate region of its corresponding
+        family of brain maps.
+
+    References
+    ----------
+    * Abraham et al. "Region segmentation for sparse decompositions: better
+      brain parcellations from rest fMRI", Sparsity Techniques in Medical Imaging,
+      Sep 2014, Boston, United States. pp.8
+    """
+    def __init__(self, n_regions=None, mask=None, target_affine=None,
+                 target_shape=None, standardize=False, low_pass=None,
+                 high_pass=None, t_r=None, memory=Memory(cachedir=None),
+                 min_size=20, threshold_strategy='voxelratio', threshold='auto',
+                 extractor='voxel_wise', smooth_fwhm=6., verbose=0):
+        self.n_regions = n_regions
+        self.mask = mask
+        self.target_affine = target_affine
+        self.target_shape = target_shape
+        self.standardize = standardize
+        self.low_pass = low_pass
+        self.high_pass = high_pass
+        self.t_r = t_r
+        self.memory = memory
+
+        self.min_size = min_size
+        self.threshold_strategy = threshold_strategy
+        self.threshold = threshold
+        self.extractor = extractor
+        self.smooth_fwhm = smooth_fwhm
+        self.verbose = verbose
+
+    def fit(self, maps_img):
+        """ Compute the mask and fit the mask to the maps data and
+        and extract or seperate the regions into region from the maps.
+
+        Parameters
+        ----------
+        maps_img: a Niimg-like image/object.
+            the image which consists of atlas maps or statistically
+            estimated maps.
+
+        Returns
+        -------
+        regions_: 4D Niimg-like image/objects.
+            a list of seperated region converted to Niimg concatenated
+            in a 4th dimension.
+
+        index_: a numpy array
+            a list of indices in an array to see the identity of each
+            seperated region to a family of atlas maps.
+        """
+        maps_img = check_niimg(maps_img)
```

Comment:
If you input 3D image, the line below will an exception. You can use `check_niimg(maps_img, atleast_4d=True)`. We need to discuss this but it's a minor detail, do not bother.